File: CountNumberswithUniqueDigits.cpp
#include<iostream>
#include<math.h>

using namespace std;

class Solution {
public:
    int countNumbersWithUniqueDigits(int n) {
    	
    	if(n == 0) return 1;
    	
    	int s = 0;
    	
    	for(int i = 1; i <= n; i++) s += count(i);
    	
    	return s;
    	
    }
    
    int count(int k) {
    	if(k <= 0) return 0;
    	if(k == 1) return 10;
    	int s = 1;
    	for(int i = 1; i < k; i++) s *= (10 - i);
		s *= 9;
		return s; 
	}
};

int main() {
	
	int n;
	cin>>n;
	
	Solution *solution = new Solution();
	cout<<solution->countNumbersWithUniqueDigits(n);
	
	return 0;
	
}

File: CountNumberswithUniqueDigits.py
class Solution:
    def countNumbersWithUniqueDigits(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n == 0:
            return 1

        # f(x) = 9 * 9 * 8 * ... * (9 - x + 2)

        s = 10
        d = 9
        for i in range(2, n + 1):
            d *= (11 - i)
            s += d
        return s


if __name__ == '__main__':
    n = int(input())
    solution = Solution()
    print(solution.countNumbersWithUniqueDigits(n))
